### ISSUE TITLE
Rule Change: new ignoredLabels option for no-missing-label-refs

### ISSUE THREAD

**Issue by crimsonjay0** [eslint/markdown#449]
### What rule do you want to change?

no-missing-label-refs

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```markdown
- [ ] Unchecked  
- [x] Checked  
- [-] In progress
```

### What does the rule currently do for this code?

In my use case, I use GFM task lists with custom states, such as [-] to indicate "in progress".
Currently, the rule reports an error for [-]: `Label reference '-' not found.` This is not helpful, since [-] is not intended to be a reference label, but a visual indicator for task status.

### What will the rule do after it's changed?

With the following configuration: `["error", { "ignoredLabels": ["-"] }]`
The rule will ignore [-] and not report it as a missing label reference.

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by snitin315** [comment]
I think that makes sense. Would like to know what the rest of the team thinks. @eslint/eslint-team

**Comment by snitin315** [comment]
Just saw that markdownlint also has the  [`ignored_labels` option](https://github.com/DavidAnson/markdownlint/blob/main/doc/md052.md) for the same use case.

**Comment by lumirlumir** [comment]
This issue is slightly different in context, but I believe it aligns with https://github.com/eslint/markdown/issues/294.

The idea of adding an `ignored_labels` (or `allowedLabels`) option to no-missing-label-refs was also discussed in https://github.com/eslint/markdown/issues/294.

In my opinion, it would be good to discuss this issue together with https://github.com/eslint/markdown/issues/294.

**Comment by JoshuaKGoldberg** [comment]
> [-] is not intended to be a reference label, but a visual indicator for task status.

🤔 is there precedent for this anywhere? I've never seen this syntax used before, and it doesn't seem supported by GFM.

`[-]` on its own is a valid label ref, even inside a list item:

- [-]

My understanding is that the rule is actually working as intended.

[-]: https://eslint.org

**Comment by lumirlumir** [comment]
@jaymarvelz Thanks for sharing your thoughts!

Right now, we're gathering feedback on the [RFC](https://github.com/eslint/rfcs/pull/138) regarding how to address this issue, so please feel free to share your thoughts on the RFC or join the existing discussions there.

Your input will be valuable when we decide on the final approach here :)

**Comment by lumirlumir** [comment]
To the team: I'm moving this issue to the 'RFC opened' column, as there is an ongoing [RFC](https://github.com/eslint/rfcs/pull/138) related to this issue.
